C++/Makefile

```diff
@@ -1,5 +1,11 @@
 # c++ version
+GCC_MAJOR:=$(shell echo $(GCC_VERSION) | cut -d'.' -f1)
+
+ifeq ($(GCC_MAJOR),5)
 CXXFLAGS += -std=c++14
```

Comment:
This project will also compile on platforms where GCC may not be installed, and Clang will be used there. So you would need to add more conditions here to first check which compiler is being used, which will be the value of `CXX` environment variable. If `CXX` turns out to be `g++` then you would check the major version.

Also, you would need to change the condition to the equivalent of `$GCC_MAJOR >= 5` because there are versions 6 and 7 of GCC too (I use 7.1.1).